### PR TITLE
Add provider tests and fix S3 secret issue

### PR DESCRIPTION
**Pull request by mbillow**
There was an attribute left out of my initial merge request (#1139) that is preventing the complete setup of resources with those changes. Fixing that error and adding tests to catch similar issues in the future.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Relates #1139

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note
BUGS:
- `resource/vault_raft_snapshot_agent_config`: Fixed issue where AWS Secret Access Key was ignored.
```

Output from acceptance testing:

```
$ TESTARGS="--run RaftSnapshotAgentConfig" TF_ACC_ENTERPRISE="true" make testacc
...
=== RUN   TestAccRaftSnapshotAgentConfig_basic
--- PASS: TestAccRaftSnapshotAgentConfig_basic (1.46s)
=== RUN   TestAccRaftSnapshotAgentConfig_import
--- PASS: TestAccRaftSnapshotAgentConfig_import (0.48s)
PASS
```

### CODE REVIEW

**Comment by mbillow** [comment]
@vinay-gopalan Hey! 😄 If you could take a quick look at this, it is just a field I forgot in my initial PR. I added some tests to make sure this doesn't happen again.

File: vault/resource_raft_snapshot_agent_config.go
Comment (mbillow) [inline vault/resource_raft_snapshot_agent_config.go:435]: This is spooky and should probably be looked into on the Vault side. When this string isn't set, Vault returns `false` instead of `null` or an empty string like the other raft snapshot storage providers. Handling it here in the provider for now though.

Comment (vinay-gopalan) [reply vault/resource_raft_snapshot_agent_config.go:435]: Can you confirm where you're seeing this returned as `false`? I modified the if case to log the Google Endpoint in case it was non-null and didn't see any logs. The API response to the TF Provider also doesn't seem to have a `google_endpoint` field in the response from Vault: 

Screen Shot 2021-09-28 at 3 04 46 PM

**Comment by vinay-gopalan** [comment]
Hi @mbillow sorry about the delay! The PR looks good! I mainly want to confirm whether the `google_endpoint` returned as `false` fix is necessary. Any logs you can provide would be great. Thanks!

File: vault/resource_raft_snapshot_agent_config.go
Comment (vinay-gopalan) [reply vault/resource_raft_snapshot_agent_config.go:435]: Does this happen only when the storage type is `google_gcs` and the endpoint isn't specified?

**Comment by mbillow** [comment]
> I mainly want to confirm whether the google_endpoint returned as false fix is necessary.

```bash
~/docker/vault-enterprise-raft
➜ VAULT_ADDR="http://localhost:28200" vault write sys/storage/raft/snapshot-auto/config/example interval=1 retain=1 path_prefix=demo storage_type=google-gcs google_gcs_bucket=demo
Success! Data written to: sys/storage/raft/snapshot-auto/config/example

~/docker/vault-enterprise-raft
➜ VAULT_ADDR="http://localhost:28200" vault read sys/storage/raft/snapshot-auto/config/example
Key                           Value
---                           -----
file_prefix                   vault-snapshot
google_disable_tls            false
google_endpoint               false
google_gcs_bucket             demo
google_service_account_key    n/a
interval                      1
path_prefix                   demo
retain                        1
storage_type                  google-gcs

~/docker/vault-enterprise-raft
➜ VAULT_ADDR="http://localhost:28200" vault write sys/storage/raft/snapshot-auto/config/example2 interval=1 retain=1 path_prefix=demo storage_type=aws-s3 aws_s3_bucket=demo aws_s3_region=us-east-1
Success! Data written to: sys/storage/raft/snapshot-auto/config/example2

~/docker/vault-enterprise-raft
➜ VAULT_ADDR="http://localhost:28200" vault read sys/storage/raft/snapshot-auto/config/example2
Key                              Value
---                              -----
aws_access_key_id                n/a
aws_s3_bucket                    demo
aws_s3_disable_tls               false
aws_s3_enable_kms                false
aws_s3_endpoint                  n/a
aws_s3_force_path_style          false
aws_s3_kms_key                   n/a
aws_s3_region                    us-east-1
aws_s3_server_side_encryption    false
aws_secret_access_key            n/a
aws_session_token                n/a
file_prefix                      vault-snapshot
interval                         1
path_prefix                      demo
retain                           1
storage_type                     aws-s3
```

You'll notice that `google_endpoint` is `false` when unset, but `aws_s3_endpoint` is `null` when unset. According to [the docs](https://www.vaultproject.io/api-docs/system/storage/raftautosnapshots#google_endpoint), it is supposed to be an optional value of type `string`.... so `false` is definitely not right.

**Comment by vinay-gopalan** [comment]
> > I mainly want to confirm whether the google_endpoint returned as false fix is necessary.
> 
> ```shell
> ~/docker/vault-enterprise-raft
> ➜ VAULT_ADDR="http://localhost:28200" vault write sys/storage/raft/snapshot-auto/config/example interval=1 retain=1 path_prefix=demo storage_type=google-gcs google_gcs_bucket=demo
> Success! Data written to: sys/storage/raft/snapshot-auto/config/example
> 
> ~/docker/vault-enterprise-raft
> ➜ VAULT_ADDR="http://localhost:28200" vault read sys/storage/raft/snapshot-auto/config/example
> Key                           Value
> ---                           -----
> file_prefix                   vault-snapshot
> google_disable_tls            false
> google_endpoint               false
> google_gcs_bucket             demo
> google_service_account_key    n/a
> interval                      1
> path_prefix                   demo
> retain                        1
> storage_type                  google-gcs
> 
> ~/docker/vault-enterprise-raft
> ➜ VAULT_ADDR="http://localhost:28200" vault write sys/storage/raft/snapshot-auto/config/example2 interval=1 retain=1 path_prefix=demo storage_type=aws-s3 aws_s3_bucket=demo aws_s3_region=us-east-1
> Success! Data written to: sys/storage/raft/snapshot-auto/config/example2
> 
> ~/docker/vault-enterprise-raft
> ➜ VAULT_ADDR="http://localhost:28200" vault read sys/storage/raft/snapshot-auto/config/example2
> Key                              Value
> ---                              -----
> aws_access_key_id                n/a
> aws_s3_bucket                    demo
> aws_s3_disable_tls               false
> aws_s3_enable_kms                false
> aws_s3_endpoint                  n/a
> aws_s3_force_path_style          false
> aws_s3_kms_key                   n/a
> aws_s3_region                    us-east-1
> aws_s3_server_side_encryption    false
> aws_secret_access_key            n/a
> aws_session_token                n/a
> file_prefix                      vault-snapshot
> interval                         1
> path_prefix                      demo
> retain                           1
> storage_type                     aws-s3
> ```
> 
> You'll notice that `google_endpoint` is `false` when unset, but `aws_s3_endpoint` is `null` when unset. According to [the docs](https://www.vaultproject.io/api-docs/system/storage/raftautosnapshots#google_endpoint), it is supposed to be an optional value of type `string`.... so `false` is definitely not right.

Yup! Sorry I was under the impression it was sending non-null values for all requests. I confirmed with the `google_gcs` storage type what you're seeing (I was initially working with local), and will open an issue on the Vault side accordingly. Your current fix for the TF Vault Provider looks good however. Thanks for bringing this up and apologies again for missing these fields in the first review! 🙇

**Review by vinay-gopalan — APPROVED** [review]
LGTM 👍

**Comment by vinay-gopalan** [comment]
We're holding off on merging things until this PR to upgrade to TF SDK v2 goes in https://github.com/hashicorp/terraform-provider-vault/pull/1175

Will merge your PR once that goes through! Thanks :)

**Comment by mbillow** [comment]
> Thanks for bringing this up and apologies again for missing these fields in the first review!

I should have had tests for the cloud providers from the start. Feel free to pin this one squarely on my laziness. 😛 

> We're holding off on merging things until this PR to upgrade to TF SDK v2 goes in #1175

I can barely hold in my excitement for the day running TF in debug mode doesn't output 50 SDK deprecation warnings per resource. 😉 

Thanks again for reviewing.

**Comment by benashz** [comment]
@mbillow this fix will go out in the v2.24.1 release tomorrow.

**Pull request merged**
